src/core/config/dns_client.h

```diff
@@ -46,33 +46,68 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_DNS_RESPONSE_TIMEOUT
+ * @def OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
  *
- * Maximum time that DNS Client waits for response in milliseconds.
+ * Define to 1 to enable DNS based Service Discovery (DNS-SD) client.
  *
  */
-#ifndef OPENTHREAD_CONFIG_DNS_RESPONSE_TIMEOUT
-#define OPENTHREAD_CONFIG_DNS_RESPONSE_TIMEOUT 3000
+#ifndef OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
+#define OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE 1
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_DNS_MAX_RETRANSMIT
+ * @def OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_IP6_ADDRESS
  *
- * Maximum number of retransmissions for DNS client.
+ * Specifies the default DNS server IPv6 address.
+ *
+ * It MUST be a C string representation of the server IPv6 address.
+ *
+ * Default value is set to "2001:4860:4860::8888" which is the Google Public DNS Ipv6 address.
  *
  */
-#ifndef OPENTHREAD_CONFIG_DNS_MAX_RETRANSMIT
-#define OPENTHREAD_CONFIG_DNS_MAX_RETRANSMIT 2
+#ifndef OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_IP6_ADDRESS
+#define OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_IP6_ADDRESS "2001:4860:4860::8888"
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
+ * @def OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_PORT
  *
- * Define to 1 to enable DNS based Service Discovery (DNS-SD) client.
+ * Specifies the default DNS server port number.
  *
  */
-#ifndef OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
-#define OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE 1
+#ifndef OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_PORT
+#define OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_SERVER_PORT 53
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_RESPONSE_TIMEOUT
+ *
+ * Specifies the default wait time that DNS client waits for a response from server (in milliseconds).
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_RESPONSE_TIMEOUT
+#define OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_RESPONSE_TIMEOUT 3000
```

Comment:
```suggestion
#define OPENTHREAD_CONFIG_DNS_CLIENT_DEFAULT_RESPONSE_TIMEOUT 6000
```

The 3s timeout seems too small. For example, `RFC8766 Discovery Proxy` sets the timeout to six seconds. So, the DNS client should use a longer timeout to better with DNS-SD server with Discovery Proxy. The default timeout of `dig` command is 5 seconds. 

Thoughts?

Reply:
I am fine changing it to 6 seconds. 

But note that there is also by default 3 tx attempts before reporting failure, so with 6 seconds wait per attempt, total wait time is 6x3 = 18 seconds.

Reply:
> But note that there is also by default 3 tx attempts before reporting failure, so with 6 seconds wait per attempt, total wait time is 6x3 = 18 seconds. I think should be fine.

I can see that. Actually that's the reason why `dig`'s 5s timeout is also fine. Thanks.